### PR TITLE
CHAOSS Pull Request for CHAOSS: Improving Open Source Sustainability

### PR DESCRIPTION
**Pull request by sgoggins**
Submitted on behalf of the CHAOSS project, Sean Goggins, Matt Germonprez, and Elizabeth Barron

### CODE REVIEW

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md   [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+Open source foundations want to identify and respond to community needs, evaluate the impact of their work, and promote communities.
+
+In response, CHAOSS was founded as a project at the Linux Foundation in 2017. Over these five years, CHAOSS has improved how all understand open source project health and sustainability in key ways, helping relevant stakeholders make more informed decisions about open source project engagement and community development. More information on CHAOSS is available at: https://chaoss.community. 
+
+With respect to the current proposal, we aim to advance three key projects with CHAOSS: (1) developing CHAOSS software to map open source software ecosystems and identify areas of critical concern within a specified ecosystem, (2) advancing CHAOSS diversity, equity, and inclusion (DEI) metrics and programs to help open source projects best center DEI in their own efforts, and  (3) building collections of CHAOSS metrics (i.e., metrics models) and making these collections deployable through CHAOSS software.
```
Comment (epicfaace) [inline proposals/CHAOSS-Improving-Open-Source-Sustainability.md  :48]:
```suggestion
With respect to the current proposal, we aim to advance three key projects with CHAOSS: (1) developing CHAOSS software to map open source software ecosystems and identify areas of critical concern within a specified ecosystem, (2) advancing CHAOSS diversity, equity, and inclusion (DEI) metrics and programs to help open source projects best center DEI in their own efforts, and (3) building collections of CHAOSS metrics (i.e., metrics models) and making these collections deployable through CHAOSS software.
```

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md   [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+Discovery: The process of identifying related open source software components is ad-hoc, leading to an unclear picture of software present within an ecosystem.
+Health Assessment: Software can be available without a healthy, active contributor community to keep it useful and synchronized with software around it. Without sustained effort software can create problems in the supply chain of nearby open-source software projects. These characteristics include Identifying ecosystem-level characteristics that correspond with reliability, safety, security, inclusivity, and what might generally be described as health trajectories.
```
Comment (epicfaace) [inline proposals/CHAOSS-Improving-Open-Source-Sustainability.md  :54]:
```suggestion
Health Assessment: Software can be available without a healthy, active contributor community to keep it useful and synchronized with software around it. Without sustained effort software can create problems in the supply chain of nearby open-source software projects. These characteristics include identifying ecosystem-level characteristics that correspond with reliability, safety, security, inclusivity, and what might generally be described as health trajectories. 
```

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md   [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+The first part in this project will scale the dependency analysis and machine learning components of the CHAOSS/Augur software project. Specifically, we will engage the CHAOSS community in an iterative process of identifying critical sub-ecosystems in package managed libraries, core corporatized infrastructure, and scientific software. We will then scale CHAOSS/Augur’s existing machine learning algorithms to discover, assess, and identify ecosystem-level characteristics that generate an ecosystem map. The map will substantially accelerate the open-source community writ large’s ability to discover opportunities for improvement, and models of practice that are effective in similar ecosystems and sub-ecosystems.
```
Comment (epicfaace) [inline proposals/CHAOSS-Improving-Open-Source-Sustainability.md  :60]: What kind of dependency analysis does Augur already do? What data sources does it get this information from? Do you plan to integrate other dependency analysis tools / APIs or just use Augur?

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md   [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+**Project 2: Centering DEI.** A key component of CHAOSS is the development of DEI metrics to help, not only ourselves, but others best center DEI within their projects. Improving open source work and the DEI share a common geography. For open source community health, the key for aligning our efforts lies in the application of research and practice illustrating how advancing DEI within open source communities enables us to overcome key open source challenges from new perspectives. More productive teams and healthier communities will emerge through this recognition, the introduction of new points of view, and a growing understanding that open source software faces shared challenges advanced through new alignments.
```
Comment (epicfaace) [inline proposals/CHAOSS-Improving-Open-Source-Sustainability.md  :64]: How does this work intersect with / follow with CHAOSS's DEI audit in 2021?

**Comment by epicfaace** [comment]
Hi! Thanks for your proposal 👋

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md   (hunk 4 above)
Comment (germonprez) [reply proposals/CHAOSS-Improving-Open-Source-Sustainability.md  :64]: Hi @epicfaace. The DEI audit was about identifying key areas of DEI work that could be done in CHAOSS specifically. We are currently implementing those. For example, we are implementing processes by which metrics translations are done for our metrics releases (https://github.com/chaoss/translations). The proposal here is about helping to bring those recommendations forward to other open source communities. We would like to make what we learned available to help others.

**Comment by germonprez** [comment]
> Hi! Thanks for your proposal 👋

Thanks for the feedback!

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md   (hunk 3 above)
Comment (sgoggins) [reply proposals/CHAOSS-Improving-Open-Source-Sustainability.md  :60]: Augur does three types of dependency analysis: 
1. We check every packaged managed library against the most recent stable release, then we calculate "libyears" for each library". This is a CHAOSS metric that provides a useful approximation of how current the package managed libraries are in a project. 
2. We enumerate all dependencies for the 10 common languages, whether or not they are package managed. 
3. We apply the OSSF Scorecard to each repository. This particular design stores the data in a single JSON field to accommodate the ongoing development of the scorecard without having it break.

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+Graphic Designer: $80,000
+Software Community Manager: $80,000
+Graduate Student: $65,000
+Hardware for Metrics Models Deployment: $15,000 
+Hardware for Ecosystem Machine Learning Analysis $40,000
+Travel: $30,000
+Overhead costs (10%): $40,000
+
+Total:  $440,000
+
```
Comment (epicfaace) [inline proposals/CHAOSS-Improving-Open-Source-Sustainability.md:107]: Timeline and outputs (we will do XXX, YYY, etc.) and outcomes (long-term effects of work)

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+In response, CHAOSS was founded as a project at the Linux Foundation in 2017. Over these five years, CHAOSS has improved how all understand open source project health and sustainability in key ways, helping relevant stakeholders make more informed decisions about open source project engagement and community development. More information on CHAOSS is available at: https://chaoss.community. 
+
+With respect to the current proposal, we aim to advance three key projects with CHAOSS: (1) developing CHAOSS software to map open source software ecosystems and identify areas of critical concern within a specified ecosystem, (2) advancing CHAOSS diversity, equity, and inclusion (DEI) metrics and programs to help open source projects best center DEI in their own efforts, and (3) building collections of CHAOSS metrics (i.e., metrics models) and making these collections deployable through CHAOSS software.
+
+Each of the three projects leverages the CHAOSS Project's established community and technology in specific ways intended to catalyze adjacent corporate, university, and non-government organization open source software initiatives. 
+
```
Comment (sgoggins) [inline proposals/CHAOSS-Improving-Open-Source-Sustainability.md:40]: @epicfaace : This is where I put outputs and outcomes,

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+Software Community Manager: $80,000
+Graduate Student: $65,000
+Hardware for Metrics Models Deployment: $15,000 
+Hardware for Ecosystem Machine Learning Analysis $40,000
+Travel: $30,000
+Overhead costs (10%): $40,000
+
+Total:  $440,000
+
+#### Timeline
```
Comment (sgoggins) [inline proposals/CHAOSS-Improving-Open-Source-Sustainability.md:120]: @epicfaace : This is where I put the timeline.

File: proposals/CHAOSS-Improving-Open-Source-Sustainability.md (hunk 5 above)
Comment (sgoggins) [reply proposals/CHAOSS-Improving-Open-Source-Sustainability.md:120]: I put the outputs and outcomes at the end of the project summary, and the timeline down here before I read your comment, @epicfaace . I'm happy to move everything to here, but thought I would let you read it first.

**Review by epicfaace — APPROVED** [review]
Thanks for the updates!

**Pull request merged**